### DISCUSSION TITLE
Pymatgen 2022.04.19 for generating interfaces

### DISCUSSION THREAD

**Discussion by Jeff-oakley** [materialsproject/pymatgen#2504] [Q&A] [answered]
It seems that we are forced to use CoherentInterfaceBuilder starting from pymatgen 2022.04.19 instead of the old pymatgen.analysis.interface.InterfaceBuilder. However, it seems the CoherentInterfaceBuilder sets up a much more strict lattice constant matching criterion. Is that still possible for us to stretch the materials and force a coherent interface like what we did in InterfaceBuilder? I just realize CoherentInterfaceBuilder will frequently dump zero interface structures when I use get_interfaces method

I guess the key problem for me is that CoherentInterfaceBuilder now requires the input of miller indices manually. Is that possible to automatically identify the best miller indices? It seems the old pymatgen.analysis.interface.InterfaceBuilder module does it

**Comment by shyuep** [comment, marked as answer]
I think there is a SubstrateAnalyzer that allows you to specify an allowed strain? Is that what you are looking for?
